apps/prairielearn/elements/pl-matching/pl-matching.py

```diff
@@ -64,7 +68,7 @@ def transform(i, opt):
     return [transform(i, opt) for i, opt in enumerate(options_list)]
 
 
-def partition(data, pred):
+def partition(data: list[Any], pred: Callable) -> tuple[list[Any], list[Any]]:
```

Comment:
Shouldn't you use a typevar with this? Then we can put it in prairielearn.py, since there's another PR that could use this helper function (https://github.com/PrairieLearn/PrairieLearn/pull/10317)

 Data can also be any iterable